draft-ietf-tls-tls13.md

```diff
@@ -2917,28 +2906,49 @@ certificate_authorities
   root CA or for a subordinate CA; thus, this message can be used to
   describe known roots as well as a desired authorization space.  If
   the certificate_authorities list is empty, then the client MAY
-  send any certificate of the appropriate ClientCertificateType,
-  unless there is some external arrangement to the contrary.
+  send any certificate that meets the rest of the selection criteria 
+  in the CertificateRequest, unless there is some external arrangement 
+  to the contrary.
+
+certificate_extensions
+: A list of certificate extension OIDs [RFC5280] with their allowed
+  values, represented in DER-encoded format. Some certificate
+  extension OIDs allow multiple values (e.g. Extended Key Usage).
+  If the server has included a non-empty certificate_extensions list,
+  the client certificate MUST contain all of the specified extension
+  OIDs that the client recognizes. For each extension OID recognized
+  by the client, all of the specified values MUST be present in the
+  client certificate (but the certificate MAY have other values as 
+  well). However, the client MUST ignore and skip any unrecognized 
+  certificate extension OIDs. If the client has ignored some of the 
```

Comment:
Sure, if this is clearer. The idea is that the client MUST ignore OIDs in the CertificateRequest that the client does not recognize, MUST not try to match them, and MUST skip to the next OID.

From: ekr [mailto:notifications@github.com]
Sent: Wednesday, October 14, 2015 2:09 PM
To: tlswg/tls13-spec <tls13-spec@noreply.github.com>
Cc: Andrei Popov <Andrei.Popov@microsoft.com>
Subject: Re: [tls13-spec] Removing certificate_types and adding certificate_extensions (#290)


In draft-ietf-tls-tls13.md<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2ftlswg%2ftls13-spec%2fpull%2f290%23discussion_r42053523&data=01%7c01%7candrei.popov%40microsoft.com%7cfbb2d7ae9f43438387ed08d2d4dbba46%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ngmkMfse4HUdeaCjmgIYKb0LEz%2f5Ld0VHYEgC%2fIdi6M%3d>:

> -  unless there is some external arrangement to the contrary.

> +  send any certificate that meets the rest of the selection criteria

> +  in the CertificateRequest, unless there is some external arrangement

> +  to the contrary.

> +

> +certificate_extensions

> +: A list of certificate extension OIDs [RFC5280] with their allowed

> +  values, represented in DER-encoded format. Some certificate

> +  extension OIDs allow multiple values (e.g. Extended Key Usage).

> +  If the server has included a non-empty certificate_extensions list,

> +  the client certificate MUST contain all of the specified extension

> +  OIDs that the client recognizes. For each extension OID recognized

> +  by the client, all of the specified values MUST be present in the

> +  client certificate (but the certificate MAY have other values as

> +  well). However, the client MUST ignore and skip any unrecognized

> +  certificate extension OIDs. If the client has ignored some of the

Perhaps "ignore the contents of"?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2ftlswg%2ftls13-spec%2fpull%2f290%2ffiles%23r42053523&data=01%7c01%7candrei.popov%40microsoft.com%7cfbb2d7ae9f43438387ed08d2d4dbba46%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=qVBla9Su4pC2RhNfR6GK84hMXhKDbc6D6ZtAbTP8A2Q%3d>.